### ISSUE TITLE
Wireframes for Neo Gui

### ISSUE THREAD

**Issue by lock9** [neo-project/neo-gui#303]
Hi,
We are planning to make a new Neo Gui, and for that, we need wireframes.
Usually, I have a list of features to build the wireframes on top of it, but if we build this internally, we may do this without that.
I think once we agree on the interface, we can move to the implementation. 

This can be started now, we don't have to wait for #302

**Comment by wanglongfei88** [comment]
@lock9 we can start it when we are discussing the UI framework. :)

**Comment by lock9** [comment]
Hi! I'm working on it @wanglongfei88. Thanks

**Comment by wanglongfei88** [comment]
@lock9 Any progress?

**Comment by lock9** [comment]
Yes @wanglongfei88: https://xd.adobe.com/view/433a454e-c57b-4475-6d7f-5d7d3c7739e3-350b/
I believe we are only 30% to 40% done, we need more iterations and the team is still learning how Neo works.
Feel free to give any feedback, we still need some iterations to make it good.
Thanks

**Comment by wanglongfei88** [comment]
@lock9, the wireframe looks great for me. The one question for me is that should we add blockchain explorer support into neo-gui as well ?

**Comment by lock9** [comment]
@wanglongfei88  the user must be able to explore local data

**Comment by wanglongfei88** [comment]
I agree that we should not add additional database for viewing info. 

Rick, please keep us updated about the UI wireframe? At the same time, we are discussing the UI framework in another PR. After talking with lightszero, we decided to try to develop a simple application using Avalonia and Electron for testing purpose. And then choose the UI framework with better development support and experience.

**Comment by lock9** [comment]
Sure @wanglongfei88 , I feature list and plan is needed for it. I will update you with more information later this week.

**Comment by wanglongfei88** [comment]
![NEO-GUI](https://user-images.githubusercontent.com/34025001/68363420-bfd5b580-0165-11ea-813e-dd170a859a84.png)

@lock9 , Rick, this is the feature list that we created based on the functions of Neo3. Please use it as a reference when your team is creating the wireframe so that we do not have too much divergence. Hopefully it will be very helpful. Please let me know if there is any thing we need to change.

**Comment by lock9** [comment]
That will definitely help a lot, thank you!

**Comment by lock9** [comment]
Hi @wanglongfei88, this is the raw feature list, I didn't evaluate with my colleague if everything will be included, there are topics that we don't even have inside neo yet. There are topics that are neo 2 only (neo fs). 
Your feedback is appreciated. Thanks.

Neo GUI
	Wallet
		Accounts
			Export
				NEP-2
			Import
				Wif
				NEP-2
				Read-only
			Create
				Single Signature
				Multiple Signatures
				Contract Composer
					Tool
				Template
					Using time lock
			View
			Delete
		Signing
			Sign and relay
			Offline signing
			Smart Contract verification script
	Plugin System
		IUIPlugin
			Extend Menu
			Custom Window
		Change plugin config.json
		Plugin repository
			List
			Install
			Remove
	State Channel
		Start new channel
		Join channel
		Move state
		Start dispute
		End channel
		View channel data
	Settings
		Protocol configuration
			Mainnet
			Testnet
			Custom
				CRUD
			protocol.json
		Application configuration
			config.json
		Plugin settings
		RPC
			Enabled / disabled
			Endpoints
				Enable / disable endpoint
			RPC Plugin support
		Consensus
			Start consensus
			Logs
				Enable / disable
				View logs
	Tools
		Type conversion
		Script disassembling
		Logging
		Account Builder
		Script Composer
	View local data
		Blocks
		Transactions
		Contracts
			View contract
				ABI
				Manifest.json
		Smart Contract Storage
			Quick data conversor support
		Query using DB key
	Interaction
		Transfer
		Relay json
		Transfer many
			Form
			Import CSV
	UI Additions
		Node Status
			Synchronization state
			Time since last block
			CPU Load
			Memory load
			Transactions in memory pool
				View memory pool transactions
			Connected nodes
		Help menu
			Docs
			Check for updates
			Current version
	ZK
		Create claim
		Create proof
		Verify proof
	Entities
		Consensus nodes
			Current consensus nodes
				Public Keys
				Public information
			Candidates
				View
				Vote
				Apply for consensus
		Oracle nodes
			Current oracle nodes
				Public keys
				Public information
			Candidates
				View
				Vote
				Apply for oracle node
		Neo FS nodes
			Inner ring
			Outer ring
			Candidates
				View
				Vote
				Apply for outer ring node
	Neo FS
		Get Data
			Verify data integrity
		Store Data
			Configure redundancy options
			Pay
			Send data

**Comment by wanglongfei88** [comment]
@lock9 Hi, Rick. I like the feature list. They are a lot of useful functions for neo3 gui. My concern is that this feature is too comprehensive and forward-looking. For example, state channel, ZK, Oracle and FS node will not be ready at first stage. 

Maybe we should start with feature plan for the 1st version of neo-gui.  And then slowly adding new features into it when those features are ready in neo3.
